src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -242,16 +243,15 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     // Rename as little as possible.  Enforce invariants on all renames.
     type TypeOfClonedSymbol >: Null <: Symbol { type NameType = Symbol.this.NameType }
 
-    // Abstract here so TypeSymbol and TermSymbol can have a private[this] field
-    // with the proper specific type.
-    def rawname: NameType
-    def name: NameType
+    final def rawname: NameType = _rawname.asInstanceOf[NameType]
+    final def name: NameType = if (needsFlatClasses) flattenedName else _rawname.asInstanceOf[NameType]
```

Comment:
Let's try to add a faster path for `MethodSymbol` that avoids the virtual call to `flattenedName`. It's probably enough to add more flag tests to `needsFlatClasses`.

Reply:
That makes sense. I looked at `needsFlatClasses` and I don't think there's anything that calls it on a method expecting the answer to be `true`.